src/main/java/com/cleanroommc/modularui/widgets/DropDownMenu.java

```diff
@@ -0,0 +1,222 @@
+package com.cleanroommc.modularui.widgets;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+
+import org.jetbrains.annotations.NotNull;
+
+import com.cleanroommc.modularui.api.drawable.IDrawable;
+import com.cleanroommc.modularui.api.widget.IGuiAction;
+import com.cleanroommc.modularui.api.widget.IWidget;
+import com.cleanroommc.modularui.api.widget.Interactable;
+import com.cleanroommc.modularui.drawable.GuiTextures;
+import com.cleanroommc.modularui.drawable.keys.StringKey;
+import com.cleanroommc.modularui.screen.viewport.GuiContext;
+import com.cleanroommc.modularui.utils.Alignment;
+import com.cleanroommc.modularui.widget.ParentWidget;
+import com.cleanroommc.modularui.widget.ScrollWidget;
+import com.cleanroommc.modularui.widget.SingleChildWidget;
+import com.cleanroommc.modularui.widget.Widget;
+import com.cleanroommc.modularui.widget.sizer.Area;
+
+public class DropDownMenu extends SingleChildWidget<DropDownMenu> implements Interactable {
+    private static final StringKey NONE = new StringKey("None");
+    private DropDownWrapper menu = new DropDownWrapper();
+    private IDrawable arrowClosed;
+    private IDrawable arrowOpened;
+
+    public DropDownMenu() {
+        menu.setEnabled(false);
+        menu.background(GuiTextures.BUTTON);
+        child(menu);
+        setArrows(GuiTextures.ARROW_UP, GuiTextures.ARROW_DOWN);
+    }
+
+    public int getSelectedIndex() {
+        return menu.getCurrentIndex();
+    }
+
+    public DropDownMenu setSelectedIndex(int index) {
+        menu.setCurrentIndex(index);
+        return getThis();
+    }
+
+    public DropDownMenu addChoice(ButtonWidget<?> button) {
+        menu.addChoice(button);
+        return getThis();
+    }
+
+    public DropDownMenu setArrows(IDrawable arrowClosed, IDrawable arrowOpened) {
+        this.arrowClosed = arrowClosed;
+        this.arrowOpened = arrowOpened;
+        return getThis();
+    }
+
+    public DropDownMenu addChoice(IGuiAction.MouseReleased onSelect, IDrawable... drawable) {
+        ButtonWidget<?> button = new ButtonWidget<>();
+        return addChoice(button.onMouseReleased(onSelect).onMouseTapped(m -> {
+                menu.setStatus(false);
+                menu.setSelectedWidget(button);
```

Comment:
```suggestion
        return addChoice(button.onMouseReleased(m -> {
                menu.setStatus(false);
                menu.setSelectedWidget(button);
                onSelect.release(m);
```

Reply:
This explains why the clicking never worked.